solr/core/src/java/org/apache/solr/metrics/DelegateRegistryTimer.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.solr.metrics;
+
+import java.time.Duration;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+import com.codahale.metrics.Snapshot;
+import com.codahale.metrics.Timer;
+
+public class DelegateRegistryTimer extends Timer {
+
+  private final Timer primaryTimer;
+  private final Timer delegateTimer;
+
+  public DelegateRegistryTimer(Timer primaryTimer, Timer delegateTimer) {
+    this.primaryTimer = primaryTimer;
+    this.delegateTimer = delegateTimer;
+  }
+
+  @Override
+  public void update(long duration, TimeUnit unit) {
+    primaryTimer.update(duration, unit);
+    delegateTimer.update(duration, unit);
+  }
+
+  @Override
+  public void update(Duration duration) {
+    primaryTimer.update(duration);
+    delegateTimer.update(duration);
+  }
+
+  @Override
+  public <T> T time(Callable<T> event) throws Exception {
+    return primaryTimer.time(event);
```

Comment:
I think you actually want to have `this` handle calls to `time(Callabale)`, `timeSupplier(Supplier)`, `time(Runnable)`, and `Context time()` -- i.e., these four methods should _not_ be overridden.
This is because they turn around and call `update(...)` on `this`, so if you delegate the `time()` calls, the update call goes to the primaryTimer, and `this` never has the opportunity to distribute the update to primary and delegate.